pennylane/estimator/resources_base.py

```diff
@@ -0,0 +1,407 @@
+    new_dirty = wm1.dirty_qubits + wm2.dirty_qubits
+    new_budget = wm1.tight_budget or wm2.tight_budget
+    new_logic = max(wm1.algo_qubits, wm2.algo_qubits)
+
+    new_wire_manager = WireResourceManager(
+        work_wires={"clean": new_clean, "dirty": new_dirty}, tight_budget=new_budget
+    )
+
+    new_wire_manager.algo_qubits = new_logic
+    new_gate_types = _combine_dict(first.gate_types, other.gate_types)
+    return Resources(new_wire_manager, new_gate_types)
+
+
+def add_in_parallel(first: Resources, other) -> Resources:  # &
+    r"""Add two resources assuming the circuits are executed in parallel.
+
+    Args:
+        first (Resources): first resource object to combine
+        other (Resources): other resource object to combine with
+
+    Returns:
+        Resources: combined resources
+    """
+    qm1, qm2 = (first.wire_manager, other.wire_manager)
+
+    new_clean = max(qm1.clean_qubits, qm2.clean_qubits)
+    new_dirty = qm1.dirty_qubits + qm2.dirty_qubits
+    new_budget = qm1.tight_budget or qm2.tight_budget
+    new_logic = qm1.algo_qubits + qm2.algo_qubits
+
+    new_wire_manager = WireResourceManager(
+        work_wires={"clean": new_clean, "dirty": new_dirty},
+        tight_budget=new_budget,
+    )
+
+    new_wire_manager.algo_qubits = new_logic
+    new_gate_types = _combine_dict(first.gate_types, other.gate_types)
+    return Resources(new_wire_manager, new_gate_types)
+
+
+def mul_in_series(first: Resources, scalar: int) -> Resources:  # *
+    r"""Multiply the resources by a scalar assuming the circuits are executed in series.
+
+    Args:
+        first (Resources): first resource object to scale
+        scalar (int): integer value to scale the resources by
+
+    Returns:
+        Resources: scaled resources
+    """
+    qm = first.wire_manager
+
+    new_clean = qm.clean_qubits
+    new_dirty = scalar * qm.dirty_qubits
+    new_budget = qm.tight_budget
+    new_logic = qm.algo_qubits
+
+    new_wire_manager = WireResourceManager(
+        work_wires={"clean": new_clean, "dirty": new_dirty},
+        tight_budget=new_budget,
+    )
+
+    new_wire_manager.algo_qubits = new_logic
+    new_gate_types = _scale_dict(first.gate_types, scalar)
+
+    return Resources(new_wire_manager, new_gate_types)
+
+
+def mul_in_parallel(first: Resources, scalar: int) -> Resources:  # @
+    r"""Multiply the resources by a scalar assuming the circuits are executed in parallel.
+
+    Args:
+        first (Resources): first resource object to scale
+        scalar (int): integer value to scale the resources by
+
+    Returns:
+        Resources: scaled resources
+    """
+    qm = first.wire_manager
+
+    new_clean = qm.clean_qubits
+    new_dirty = scalar * qm.dirty_qubits
+    new_budget = qm.tight_budget
+    new_logic = scalar * qm.algo_qubits
+
+    new_wire_manager = WireResourceManager(
+        work_wires={"clean": new_clean, "dirty": new_dirty},
+        tight_budget=new_budget,
+    )
```

Comment:
Judging by how the wire manager is used, it isn't really a "manager" that persists and "manages" a pool of wires. It doesn't really "manage" anything. Instead, it's just a data structure that represents the state of the wires at a given level of the circuit?